stream_settings: Remove grey-box css.

YashRE42 opened the pull request:
<!-- What's this PR for?  (Just a link to an issue is fine.) -->
Fixes: #20222.

**Testing plan:** <!-- How have you tested? -->
node/puppeteer tests and screenshots.

**GIFs or screenshots:** <!-- If a UI change.  See:
  https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
  -->
Dark mode stream types modal:
![](https://user-images.githubusercontent.com/33805964/142911321-d85a92c4-ae41-4d4b-9fc8-3ad086e14ced.png)
Light mode stream types modal:
![](https://user-images.githubusercontent.com/33805964/142911394-5a8ce84a-c69c-47bf-9fbf-64ac6258bb8a.png)

Dark mode personal settings tab:
![](https://user-images.githubusercontent.com/33805964/142913716-def94769-7687-4ada-b36f-af25c371ebc7.png)
Light mode personal settings tab
![](https://user-images.githubusercontent.com/33805964/142913921-e1e63607-a9d2-4562-9696-7e31e2e35798.png)

<!-- Also be sure to make clear, coherent commits:
  https://zulip.readthedocs.io/en/latest/contributing/version-control.html
  -->

timabbott commented on static/templates/stream_settings/stream_types.hbs:
```diff
@@ -1,4 +1,4 @@
-<div class="grey-box new-style">
+<div class="grey-box">
```
`new-style` means something specific -- it refers to the last redesign we did to the settings buttons UI.  I think our goal should probably be to just have it be the default style for the application.

I think the Micromodal migration is going to end up removing a very large portion of explicit references to `new-style`, so we might be in a position to try to clean that up in a month or two.  May be worth a #frontend thread.

timabbott replied:
But I also think this suggests that it's not super constructive to rework elements to assume that the `new-style` styling isn't present, since our plan is to make that replace Boostrap as the default button (etc.) styles.

timabbott replied:
Given that it's literally one CSS block that you're adding in this commit, it's probably fine to keep things as-is if including `new-style` makes life more difficult.  

Actually I'm a bit puzzled; `static/templates/stream_settings/stream_settings_overlay.hbs` already has the `new-style` class.  Is the situation that the model doesn't inherit it, but the "stream creation form" overlay does?

timabbott commented on static/templates/stream_settings/stream_settings.hbs:
```diff
@@ -75,43 +75,39 @@
             </div>
             <div class="subscription-config">
-                <ul>
+                <div class="other_settings">
```
I think the elements this applies to are the `subsection-parent` or `subsection-header` classes from the Settings UI; it might make sense to make the two layers here match those names (also might give better names for these divs in the CSS that are more declarative).

timabbott commented:
The visuals lgtm; posted a few comments that might help simplify this further (and decrease differences in the implementation from what we have in, say, "Notification settings").  

@alya FYI to double-check the visuals.

I think we should also discuss a bit in #frontend the plans for the unrelated use of the `grey-box` class name, which is the greyb boxes for the forms to add linkifiers/custom emoji/etc.:

```
$ git grep grey-box
static/styles/app_components.css:.grey-box {
static/styles/dark_theme.css:    .grey-box,
static/styles/settings.css:.grey-box .wrapper {
static/templates/settings/alert_word_settings.hbs:        <div class="add-new-alert-word-box grey-box">
static/templates/settings/data_exports_admin.hbs:        <div class="add-new-export-box grey-box">
static/templates/settings/default_streams_list_admin.hbs:        <div class="add-new-default-stream-box grey-box">
static/templates/settings/display_settings.hbs:            <div class="emojiset_choices grey-box">
static/templates/settings/emoji_settings_admin.hbs:        <div class="add-new-emoji-box grey-box">
static/templates/settings/linkifier_settings_admin.hbs:            <div class="add-new-linkifier-box grey-box">
static/templates/settings/playground_settings_admin.hbs:            <div class="add-new-playground-box grey-box">
static/templates/settings/profile_field_settings_admin.hbs:        <div class="add-new-profile-field-box grey-box">
static/templates/settings/user_groups_admin.hbs:            <div class="add-new-user-group-box grey-box">
```

Those are one of our uglier components, and we should probably have a #design discussion about how we want to replace them.  One theory is that we should replace them all with simple modals and not have a form inline; that'd I think be more consistent with how other modern apps manage this sort of settings form.

alya commented:
Removing it in these places looks good to me.

amanagr commented:
@YashRE42 What tool are you using to generate these before-after screenshots?

YashRE42 commented:
> @YashRE42 What tool are you using to generate these before-after screenshots?

Oh, it's nothing special, I just:
- create a new image at https://pixlr.com/x/
- take the before screenshot with windows snip, paste and label it as "before" with "add new text",
- take the after screenshot with windows snip, paste and label it as "after" with "add new text",
- set zoom to 100%
- take a screenshot of the image, paste it here on the PR.

YashRE42 commented:
Ok, so,
(1) I changed the `other_settings` and `notification_settings` classes this was adding to just be `subsection-parent`, which is much better, so thanks for that suggestion.

(2) I have redone the commit that removes the grey-box from the personal tab to also correctly remove the grey box from the "create stream" tab, previously it was just breaking "create stream".

Dark mode stream creation panel:
![](https://user-images.githubusercontent.com/33805964/143004264-0c9aceb2-6924-44ce-8298-c9a8957c2274.png)

Light mode stream creation panel:
![](https://user-images.githubusercontent.com/33805964/143005067-b4e612ce-c5c0-4d90-8410-16f5bf16e4a7.png)

(3) I moved the commit that removes the new-style class to be the last commit here, so it's easier to drop if we choose to.
> Is the situation that the model doesn't inherit it, but the "stream creation form" overlay does?

Yes, precisely. The micromodal is a separate html block ie it does not nest under "subscription_overlay", hence, the `new-style` class doesn't apply to it.

 I have:
- [x] moved the "Fixes: xyz" so that it's on the last commit.
- [x] have verified that things will visually look the same, with or without that this commit.
- [x] edited the commit message to explain another change this commit causing, see the gifs bellow:

<details>
<summary>
before
</summary>

![before](https://user-images.githubusercontent.com/33805964/142990003-c04f163d-2813-4bc2-b0e7-daba86adc38b.gif)
</details>
<details>
<summary>
after
</summary>

![after](https://user-images.githubusercontent.com/33805964/142990007-a7768001-9e06-4502-8fea-e2209dc3ceea.gif)
</details>

(4) While working on (2) I noticed this on `main`:
<details>
<summary>
gif
</summary>

![](https://user-images.githubusercontent.com/33805964/143005764-19a485e3-24e7-485d-9fb8-2509f5a32be0.gif)
</details>
ie we don't show the input to set the number of days for message retention when creating a stream.

YashRE42 commented:
cc: @alya re: (2) above ^

timabbott commented on static/styles/subscriptions.css:
```diff
@@ -1019,12 +1015,12 @@
 }
 
-#subscription_overlay ul > li {
+#subscription_overlay .subsection-parent > div {
```
We never want CSS like this, that refers to a specific object just be it being a div inside some other object, because that's hard to read and does unpredictable things if we later e.g. add a tooltip or modal inside this parent.  I think you want `.input-group` instead of `> div` here, and then possibly adding the `input-group` class to the new div for the streamcolor input.

timabbott commented:
Posted one comment on the code; otherwise lgtm.  Visually, I think the ~5px left indentation for the "Stream permissions" blocks should be removed; it doesn't look great and I don't think we do similar indentation of heading anywhere else.

timabbott commented:
BTW does this fix the bug noted in (4) above?  It sounds like it does from the screencast, but wanted to confirm.

YashRE42 commented:
I added divs with the `input-group` class, removed the indentation on the stream_creation view and added two commits to fix (4). I also just kept the commit that removes "new-style" at the bottom, since it has the "Fixes" text and it would be confusing to put it on the input bug fix.

timabbott commented:
I'm taking care of the rebase past #20340.

Opened https://github.com/zulip/zulip/issues/20345, which I noticed while reviewing this.

timabbott commented:
Hmm, this seems broken: 
![image](https://user-images.githubusercontent.com/2746074/143138014-1a03aaa6-4b82-41dd-b4f8-cd39aee3f6c9.png)

timabbott commented:
OK, I added a new final commit that fixes that problem, while also deleting more custom CSS for this page where we'd be happier just doing the exact same thing we have in the main settings UI.  I've opened https://github.com/zulip/zulip/issues/20347 with another follow-up noted while working on this.

@alya it's probably worth your doing a round of clicking around "stream settings"; I suspect there may be a couple more quirky things after merging this and #20340 today.  Here's the items I've noted and hope @YashRE42 can fix tomorrow so that we'll be in a reasonably clean state:
* https://github.com/zulip/zulip/issues/20345
* https://github.com/zulip/zulip/issues/20347
* I think we want to come up with a more designed view of how the "Stream color" input should look.

But overall I think this is great progress, since it moves us to a world where we don't have two independent systems for trying to make checkboxes implemented in the CSS.  Also @m-e-l-u-h-a-n FYI since this pull request in a way completes one of the big transitions you started this summer :).

alya commented:
If the follow-ups are happening ~tomorrow, can you guys ping me when they are done so that I don't have to try to figure out whether we already know about the issues I see? I can take a look at that time.
